src/index.tsx

```diff
@@ -20,122 +25,136 @@ export default class SignatureCanvas extends Component {
     const copy = document.createElement('canvas')
     copy.width = this._canvas.width
     copy.height = this._canvas.height
-    copy.getContext('2d').drawImage(this._canvas, 0, 0)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    copy.getContext('2d')!.drawImage(this._canvas, 0, 0)
     // then trim it
     return trimCanvas(copy)
   }
 
   // return the internal SignaturePad reference
-  getSignaturePad = () => {
+  getSignaturePad = (): SignaturePad => {
     return this._sigPad
   }
 
-  _checkClearOnResize = () => {
+  _checkClearOnResize = (): void => {
     if (!this.props.clearOnResize) {
       return
     }
     this._resizeCanvas()
   }
 
-  _resizeCanvas = () => {
-    const canvasProps = this.props.canvasProps || {}
+  // a few eslint disables of strict-boolean-expressions here because changing
+  // these lines would be breaking -- 0s, '', null etc are falsey
+  _resizeCanvas = (): void => {
+    const canvasProps = this.props.canvasProps ?? {}
     const { width, height } = canvasProps
     // don't resize if the canvas has fixed width and height
-    if (width && height) {
+    if (typeof width !== 'undefined' && typeof height !== 'undefined') {
       return
     }
 
     const canvas = this._canvas
     /* When zoomed out to less than 100%, for some very strange reason,
       some browsers report devicePixelRatio as less than 1
       and only part of the canvas is cleared then. */
-    const ratio = Math.max(window.devicePixelRatio || 1, 1)
+    const ratio = Math.max(window.devicePixelRatio ?? 1, 1)
 
-    if (!width) {
+    if (typeof width === 'undefined') {
       canvas.width = canvas.offsetWidth * ratio
     }
-    if (!height) {
+    if (typeof height === 'undefined') {
       canvas.height = canvas.offsetHeight * ratio
     }
-    canvas.getContext('2d').scale(ratio, ratio)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    canvas.getContext('2d')!.scale(ratio, ratio)
     this.clear()
   }
 
-  render () {
+  render: Component['render'] = () => {
     const { canvasProps } = this.props
-    return <canvas ref={(ref) => { this._canvas = ref }} {...canvasProps} />
+    return <canvas ref={this._setRef} {...canvasProps} />
   }
 
   // all wrapper functions below render
   //
-  on = () => {
+  on: SignaturePad['on'] = () => {
     window.addEventListener('resize', this._checkClearOnResize)
     return this._sigPad.on()
   }
 
-  off = () => {
+  off: SignaturePad['off'] = () => {
     window.removeEventListener('resize', this._checkClearOnResize)
     return this._sigPad.off()
   }
 
-  clear = () => {
+  clear: SignaturePad['clear'] = () => {
     return this._sigPad.clear()
   }
 
-  isEmpty = () => {
+  isEmpty: SignaturePad['isEmpty'] = () => {
     return this._sigPad.isEmpty()
   }
 
-  fromDataURL = (dataURL, options) => {
+  fromDataURL: SignaturePad['fromDataURL'] = (dataURL, options) => {
     return this._sigPad.fromDataURL(dataURL, options)
   }
```

Comment:
Passing the args can be simplified by using rest/spread operator:
(also, `return` and braces can be ommited)

```js
fromDataURL: SignaturePad['fromDataURL'] = (...args) => this._sigPad.fromDataURL(...args)
```

Reply:
Stylistic changes are off-topic for this PR -- perhaps you didn't notice but the changes here are strictly isolated to that which _needs_ to change.
- I also think code formatting should remain consistent and shouldn't normally be changed (using `standard` for that reason).
- This would be too long to fit on one line anyway. And I was trying to keep the style consistent in all these methods as well. And consistent with the docs.
- Notably this is also a semantic change that I wouldn't want to make on a whim either (and is similarly off-topic to this PR).

<hr>

I welcome your review @ksocha , but please avoid making off-topic/stylistic comments. It really detracts heavily from this PR especially as roughly half of your comments have been off-topic. Please bear in mind that I'm volunteering time to read & respond to these.

Also, I would hope you didn't mean it intentionally, but your tone sometimes comes off as a bit overbearing -- for the most part, I've considered the options and written the code intentionally so. My suggestion would be to word some of these comments as questions rather than statements.